src/library/scanner/libraryscanner.cpp

```diff
@@ -170,6 +156,17 @@ void LibraryScanner::run() {
 
 void LibraryScanner::slotStartScan() {
     qDebug() << "LibraryScanner::slotStartScan";
+    DEBUG_ASSERT(m_state == STARTING);
```

Comment:
If we hit that we have an compile time error, there is nothing to handle. Returning here in non debug will make the things worse.